projects/website/src/doxygen/XSDToDoxygen.py

```diff
@@ -0,0 +1,284 @@
+                            self.name.append(i.getAttribute("name"))
+                            fout.write(" * @{")
+                            fout.write("\n")
+                            self.parse(i,fout)
+                            if len(self.writer)>0:
+                                fout.write(self.writer)
+                                self.writer=""
+                            fout.write(" * @}")
+                            fout.write("\n")
+
+            elif i.nodeName=="xs:sequence":
+                """
+                Case6:
+                if nodename==sequence
+                    then simply parse again for child nodes
+                """
+
+                self.parse(i,fout)
+            
+            elif i.nodeName=="xs:attribute":
+                """
+                Case7:
+                if nodeName==attribute
+                    Then write a specific defgroup and add comment
+                """
+                if len(self.writer)>0:
+                    fout.write(self.writer)
+                    self.writer=""
+                fout.write(" * @defgroup CDM_"+self.name[-1]+"_"+i.getAttribute("name")+" "+i.getAttribute("name"))     
+                fout.write("\n")
+            
+            elif i.nodeName == "xs:include":
+                """
+                Case8:
+                    if nodeName==include
+                    Stores all those file names into the files to be processed list
+                """
+                files=i.getAttribute("schemaLocation")
+                if files.startswith("./"):
+                    files=os.path.dirname(self.files_processed[0])+files[1:]
+                elif files.startswith(".."):
+                    files=os.path.dirname(self.files_processed[0])+files[2:]
+                else:
+                    if files=="CommonDataModel.xsd":
+                        files=os.path.dirname(self.files_processed[0])+"/"+files
+                    else:
+                        files=os.path.dirname(self.files_processed[0])+"/cdm/"+files
+                if files not in self.files_processed:
+                    self.files_processed.append(files)
+
+            else:
+                self.parse(i,fout)
+    
+if __name__=="__main__":
+    parser = argparse.ArgumentParser(description='Creation of xsd.ixx file after parsing all xsd files ')
+    parser.add_argument('-p','--xsdpath',help='Path to BioGearsDataModel.xsd file',required=True)
+    parser.add_argument('-f','--filepath',help='Name and path of .ixx file created',required=True)
+    args=parser.parse_args()
+    if args.xsdpath ==None and args.filepath ==None:
+        parser.print_help()
+        sys.exit(0)
+    else:
+        XSD=XSDToDoxygen()
+        XSD.files_processed.append(args.xsdpath)
+        destDir=args.filepath
+        if not os.path.exists(destDir):
+            os.mkdir(destDir)
+        fout=open(os.path.join(destDir,"xsd.ixx"),'w',encoding="utf-8")
+        fout.write("/**")
```

Comment:
So this is a little ugly and it distracts from what you are trying to do. String literals in most languages auto concat to the immediate proceeding string literal allowing you to format them more cleanly then this. 

    print("There is a bear here\n"
          "The bear has a bunch of honey\n"
          "The fat bear is in front of another door\n"
          "How are you going to move the bear?"
    )

Reply:
Ok sure I will change that to single line writing part :+1: